specutils/manipulation/resample.py

```diff
@@ -0,0 +1,409 @@
+        fin_lamb : ndarray
+            The desired dispersion array.
+
+        Returns
+        -------
+        resample_spectrum : `~specutils.Spectrum1D`
+            An output spectrum containing the resampled `~specutils.Spectrum1D`
+        """
+        out_flux = self._interpolation(orig_spectrum.spectral_axis, orig_spectrum.flux,
+                                  fin_lamb)
+
+        # todo: for now, use the units from the pre-resampled
+        # spectra, although if a unit is defined for fin_lamb and it doesn't
+        # match the input spectrum it won't work right, will have to think
+        # more about how to handle that... could convert before and after
+        # calculation, which is probably easiest. Matrix math algorithm is
+        # geometry based, so won't work to just let quantity math handle it.
+        # todo: handle uncertainties for interpolated cases.
+        resampled_spectrum = Spectrum1D(flux=out_flux * orig_spectrum.flux.unit,
+                                        spectral_axis=np.array(fin_lamb) * orig_spectrum.spectral_axis_unit)
+
+        return resampled_spectrum
+
+
+class SplineInterpolatedResample(ResampleBase):
+    """
+    This resample algorithim uses a cubic spline interpolator.  In the future
+    this can be expanded to use splines of different degrees.
+
+    Examples
+    --------
+
+    To resample an input spectrum to a user specified dispersion grid using
+    a cubic spline interpolator:
+
+    >>> import numpy as np
+    >>> import astropy.units as u
+    >>> from specutils import Spectrum1D
+    >>> from specutils.manipulation import SplineInterpolatedResample
+    >>> input_spectra = Spectrum1D(
+    ...     flux=np.array([1, 3, 7, 6, 20]) * u.mJy,
+    ...     spectral_axis=np.array([2, 4, 12, 16, 20]) * u.nm)
+    >>> resample_grid = np.array([1, 5, 9, 13, 14, 17, 21, 22, 23])
+    >>> fluxc_resample = SplineInterpolatedResample()
+    >>> output_spectrum1D = fluxc_resample(input_spectra, resample_grid) # doctest: +IGNORE_OUTPUT
+
+    """
+    def __init__(self, bin_edges='nan_fill'):
+        super().__init__(bin_edges)
+
+    def __call__(self, orig_spectrum, fin_lamb):
+        """
+        Return the resulting `~specutils.Spectrum1D` of the resampling.
+        """
+        return self.resample1d(orig_spectrum, fin_lamb)
+
+    def _interpolation(self, orig_dispersion, flux, fin_lamb):
+        """
+        Use specified interpolation to calculated resampled
+        flux.
+
+        Parameters
+        ----------
+        orig_dispersion : ndarray
+            The original dispersion array.
+        flux: ndarray
+            The flux array from the input Spectrum1D
+        fin_lamb : ndarray
+            The desired dispersion array.
+
+        Returns
+        -------
+        resample_flux : ndarray
+            The resampled flux array generated from the interpolation.
+
+        """
+        cubic_spline = CubicSpline(orig_dispersion, flux, extrapolate=False)
```

Comment:
Can you clarify why you used this instead of a `scipy.interpolate.InterpolatedUnivariateSpline`?  I just did some tests now and it seems like `InterpolatedUnivariateSpline` is a bit faster (and has the added advantage that you can set the spline degree as a free parameter).

Reply:
I probably just didn't look into it enough when I was putting this class together. I will switch it out and add the degree as a class parameter.

Reply:
So this gets tricky in that right now we decided offline to only do NaN fill on the boundaries for now. This is possible using CubicSpline very easy.  But it's not easy to do with UnivariateSpline.  There's no built in way, so we'd have to do a more complicated work around, which I'm not sure I should spend the time on?  Given the situation, I'm for keeping it as a CubicSpline so that the edge handling is the same for all resampling, and change it over to UnivariateSpline when there is more time.